src/classes/UTIL_Jwt.cls

```diff
@@ -0,0 +1,241 @@
+    BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+    CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+    LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+/**
+* @author Salesforce.org
+* @date 2020
+* @group Utilities
+* @description Class which holds static methods for generating jwts
+*/
+public with sharing class UTIL_Jwt {
+    // error messages consts
+    @TestVisible private static final String INVALID_BASE64_MESSAGE = 'The base64 format is invalid. ' +
+            'The number of characters must be a multiple of 4 and can ' +
+            'only contain these characters: [a-z A-Z 0-9 + / =]';
+    @TestVisible private static final String INVALID_BASE64URL_MESSAGE = 'The base64url format is invalid. ' +
+            'It can only contain these characters: [a-z A-Z 0-9 _ -]';
+    @TestVisible private static final String INVALID_PAYLOAD_MESSAGE = 'When signing JWT, ' +
+            'the Payload must contain a value.';
+    @TestVisible private static final String INVALID_KEY_MESSAGE = 'When signing JWT, ' +
+            'the Key must contain a value and use base64 format.';
+
+    // private consts
+    private static final String BASE64_REGEX = '^[a-zA-Z0-9\\+/]*={0,3}$';
+    private static final String BASE64URL_REGEX = '^[a-zA-Z0-9_-]*$';
+    private static final String ENC_ALG = 'HS256';
+    private static final String ENC_TYPE = 'JWT';
+    private static final String ALG_NAME = 'HmacSHA256';
+
+    // custom exception
+    public class UTIL_JwtException extends Exception {}
+
+    /**
+     * @description Returns the compact serialization representation of the jwt as a concatenated string,
+     *          with the encoded jwt header, encoded jwt payload, and encoded jwt signature strings
+     *          separated by period ('.') characters
+     * @param   Auth.JWT jwtInstance, instance
+     * @param   String key, base64 encoded key
+     * @return  String
+     */
+    public static String getSignedJWT(Auth.JWT jwtInstance, String key) {
+
+        if (jwtInstance==null) {
+            throw new UTIL_JwtException(INVALID_PAYLOAD_MESSAGE);
+        }
+
+        return getSignedJWTFromString(jwtInstance.toJSONString(), key);
+
+    }
+
+    /**
+     * @description helper method to sign jwt using payload as string and base64 format key
+     * @param   String payload, jwt payload
+     * @param   String key, base64 encoded key
+     * @return  String, returns the compact serialization representation of the jwt as a concatenated string,
+     *          with the encoded jwt header, encoded jwt payload, and encoded jwt signature strings
+     *          separated by period ('.') characters.
+     */
+    public static String getSignedJWTFromString(String payload, String key) {
```

Comment:
I would like to see detailed in the comments an example of the payload with some claims in it.